File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/fbos/ReadOnlyFbo.java
package org.saar.lwjgl.opengl.fbos;

import org.saar.lwjgl.opengl.fbos.exceptions.FrameBufferException;

public interface ReadOnlyFbo {

    /**
     * Returns the width of the fbo
     *
     * @return the width
     */
    int getWidth();

    /**
     * Returns the height of the fbo
     *
     * @return the height
     */
    int getHeight();

    /**
     * Bind the fbo
     */
    void bind();

    /**
     * Unbind the fbo
     */
    void unbind();

    /**
     * Ensure that the fbo status is good
     */
    void ensureStatus() throws FrameBufferException;
}

File: pe-lwjgl-binding/src/main/java/org/saar/lwjgl/opengl/fbo/ReadOnlyFbo.kt
package org.saar.lwjgl.opengl.fbo

import org.saar.lwjgl.opengl.fbo.exceptions.FrameBufferException
import org.saar.lwjgl.opengl.utils.GlBuffer

interface ReadOnlyFbo {
    /**
     * Bind the fbo
     */
    fun bind()

    /**
     * Unbind the fbo
     */
    fun unbind()

    /**
     * Set as read fbo
     */
    fun bindAsRead()

    /**
     * Set as read fbo
     */
    fun bindAsDraw()

    /**
     * Blit the fbo into the bound read fbo
     */
    fun blitFramebuffer(
        x1: Int, y1: Int, w1: Int, h1: Int, x2: Int, y2: Int, w2: Int,
        h2: Int, filter: FboBlitFilter, buffers: Array<out GlBuffer>
    )

    /**
     * Ensure that the fbo status is good
     */
    @Throws(FrameBufferException::class)
    fun ensureStatus()
}
